server/data/data-ldap/src/main/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepository.java

```diff
@@ -388,6 +388,20 @@ public void configure(HierarchicalConfiguration configuration) throws Configurat
         filter = configuration.getString("[@filter]");
 
         administratorId = Optional.fromNullable(configuration.getString("[@administratorId]"));
+
+        checkState();
+    }
+
+    private void checkState() throws ConfigurationException {
+        if (userBase == null) {
```

Comment:
I want to throw ConfigurationException to be more consistant with the type of exception thown in othe places